futures-util/src/async_await/random.rs

```diff
@@ -17,7 +17,7 @@ pub fn shuffle<T>(slice: &mut [T]) {
 }
 
 /// Return a value from `0..n`.
-fn gen_index(n: usize) -> usize {
+pub fn gen_index(n: usize) -> usize {
```

Comment:
I needed an rng for the 'delayed streams' tests. `pub(crate)` didn't seem to work for that, I guess because this is in futures-util and the tests are all in futures.